### PR TITLE
Remove inifinite loop on actionObjectUpdateAfter hook

### LINKED ISSUE #29560: Hook actionObjectUpdateAfter - Infinite loop that leads to memory limit in FO
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Hi,
i don't understand why the core is doing this but i'm facing an infinite loop when customer is loggedin but its totally logical if I read the code. First it was working but since I hooked a module on actionObjectUpdateAfter i get this syndrom.

I searched in code and this is what i found

When we are logged in as customer on front, 
It calls 
1. Hook::exec 
which check user logged in groups in Hook class with the method 
2. getHookModuleExecList who calls
3. getAllHookRegistrations.
This function calls 
4. $customer->isLogged() 
and this function calls 
5. Context::getContext()->cookie->isSessionAlive()
this function calls (on cookie class)
6. $this->getSession($this->session_id)
and the getSession method calls 
7. $session->save() 
who calls in object model the 
8. update() method who calls the 
9. hook actionObjectUpdateAfter who tried to Hook::exec 
then restart this infinite loop (back to 1.) .

I succed to stop this look by adding in objectModel line 864 

```
if(get_class($this) != 'EmployeeSession' && get_class($this) != 'CustomerSession'){
	Hook::exec('actionObjectUpdateAfter', ['object' => $this]);
	Hook::exec('actionObject' . $this->getFullyQualifiedName() . 'UpdateAfter', ['object' => $this]);
}
```
But with this it breaks the hook actionObjectUpdateAfter on EmployeeSession class and CustomerSession which extends ObjectModel too...

Edit : I figured out that natively (I mean when fresh install prestashop) this hook is not in database and is added only once a module is attached to this hook ( actionObjectUpdateAfter ) si until the hook doesn't appear in ps_hook table there is not this issue but when the row exists in ps_hook we got an infinite loop....

### Expected behavior

We can't have an infinite loop...

### Steps to reproduce

1. I attached a module on actionObjectUpdateAfter but the module is uninstalled since (i don't know if it is linked to this i checked on DB there is no more entries of attached module on this hook) ...
2. I logged in in front and i get a max execution time with a random location of killing process.. (it's never the same end position)

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

8.0

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by lartist**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  8.0.x 
| Description?      | remove inifinite loop on actionObjectUpdateAfter hook
| Type?             | bug fix 
| Category?         | FO 
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29560
| Related PRs       | 
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/29560
| Possible impacts? |

### CODE REVIEW

File: classes/Cookie.php [hunk 1]
```diff
@@ -86,6 +86,8 @@ class CookieCore
     /** @var bool */
     protected $_secure = false;
 
+    protected $session = null;
```
Comment (jolelievre) [inline classes/Cookie.php:90]: Can you please add a PHPDoc for this field?

**Review by jolelievre — APPROVED** [review]
Thanks @lartist

**Review by kpodemski — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @lartist @PrestaShop/prestashop-maintainers ,

I'm trying to reproduce the [original issue](https://github.com/PrestaShop/PrestaShop/issues/29560); but I can't anymore.
No more error/fatal error on FO.

Also tested by @khouloudbelguith :
Installed this module https://github.com/PrestaShop/ps_qualityassurance & added this hook `actionObjectUpdateAfter` 
image
image (1)

Is the PR still relevant ?
Should it be closed ?

Thanks!

**Review by matks — COMMENTED** [review]
> I'm trying to reproduce the https://github.com/PrestaShop/PrestaShop/issues/29560; but I can't anymore.
No more error/fatal error on FO.

Ohohoh 😄 the bug was solved with no modification from our part? that is a good news

But I'm worried anyway XD @lartist do you think you can try to reproduce the initial bug with the latest changes on 8.0.x ? It's possible some other pull request we merged had a side-effect of fixing this bug. We need to be sure.

**Comment by lartist** [comment]
I'm on it

**Comment by lartist** [comment]
Very weird @khouloudbelguith, I mount a shop from a new 8.0.x branch and i did the same thing than you withe the https://github.com/PrestaShop/ps_qualityassurance module ant this what I get:
Capture d’écran 2022-09-30 à 10 47 35

Did you logged into the front ?

**Review by florine2623 — COMMENTED** [review]
Hello @lartist ,

I tested again.
I installed a fresh install of 8.0.x.

I added the hook manually in database AND also via ps_qualityassurance module and in both cases I have access to my FO and my BO without error.

https://user-images.githubusercontent.com/16019289/194560014-1b7d541c-a171-4389-a63e-a95370d3cdb6.mov

I'm not sure why I can't reproduce it 🤔 


Also do not hesitate to remove the `waiting for author` label and add the `waiting for QA` when you answer ^^

**Comment by lartist** [comment]
> Hello @lartist ,
> 
> I tested again. I installed a fresh install of 8.0.x.
> 
> I added the hook manually in database AND also via ps_qualityassurance module and in both cases I have access to my FO and my BO without error.
> 
>  Screen.Recording.2022-10-07.at.14.59.59.mov 
> I'm not sure why I can't reproduce it 🤔
> 
> Also do not hesitate to remove the `waiting for author` label and add the `waiting for QA` when you answer ^^

Hi @florine2623, I dont think I can change labels since I don't have commiter's right yet.

If you have any time we may see this bug in live today.

**Review by florine2623 — APPROVED** [review]
Hi @lartist ,

Tested it once more time ^^
I needed to do some more actions before seeing the fatal error. (order, product creation, catalog mode, etc)

The error is the following when I try to finish the checkout process:
Screenshot 2022-10-17 at 17 06 36

Tested the same actions with the PR, it is looking good ^^
No more fatal error. I can finish the checkout process and make an order without issue. 
It is QA ✅ 

Thank you for your patience !

**Comment by matks** [comment]
Thank you @lartist

**Comment by PrestaEdit** [comment]
I know that is already merged, but just tested and work well, thanks for this one ! :)

**Pull request merged**